libp2p/security/insecure/transport.py

```diff
@@ -1,12 +1,37 @@
+from libp2p.crypto.keys import PublicKey
 from libp2p.network.connection.raw_connection_interface import IRawConnection
 from libp2p.peer.id import ID
 from libp2p.security.base_session import BaseSession
 from libp2p.security.base_transport import BaseSecureTransport
 from libp2p.security.secure_conn_interface import ISecureConn
+from libp2p.typing import TProtocol
+from libp2p.utils import encode_fixedint_prefixed, read_fixedint_prefixed
+
+from .exceptions import UpgradeFailure
+from .pb import plaintext_pb2
+
+# Reference: https://github.com/libp2p/go-libp2p-core/blob/master/sec/insecure/insecure.go
+
+
+PLAINTEXT_PROTOCOL_ID = TProtocol("/plaintext/2.0.0")
 
 
 class InsecureSession(BaseSession):
-    pass
+    # FIXME: Update the read/write to `BaseSession`
+    async def run_handshake(self):
+        msg = make_exchange_message(self.local_private_key.get_public_key())
+        msg_bytes = msg.SerializeToString()
+        encoded_msg_bytes = encode_fixedint_prefixed(msg_bytes)
+        self.writer.write(encoded_msg_bytes)
+        await self.writer.drain()
```

Comment:
FYI, if you *think* that you might have multiple code paths concurrently calling this you might need to wrap this in a lock.  No need to do so pre-emptively but wanted to alert you to this.

see: https://github.com/ethereum/lahja/blob/93610b2eb46969ff1797e0748c7ac2595e130aef/lahja/asyncio/endpoint.py#L99-L102

Reply:
Really appreciate that you bring this up! Since write is changed to an [async function in Python 3.8](https://docs.python.org/3.8/library/asyncio-stream.html#asyncio.StreamWriter.write), I will leave the pattern "write and drain" unchanged and add locks to these places.